### PR TITLE
Provide the behaviour the Docker images patch into us by ourselves

### PR DESCRIPTION
**Pull request by Al2Klimov**
to make that patch obsolete.

CC @julianbrost FYI

## Before

Bildschirm­foto 2023-01-23 um 15 51 06
Bildschirm­foto 2023-01-23 um 15 52 19

## After

Bildschirm­foto 2023-01-23 um 15 43 06
Bildschirm­foto 2023-01-23 um 15 46 20

### CODE REVIEW

File: application/forms/Config/General/LoggingConfigForm.php [hunk 1]
```diff
@@ -38,12 +54,7 @@ public function createElements(array $formData)
                 'autosubmit'    => true,
                 'label'         => $this->translate('Logging Type'),
                 'description'   => $this->translate('The type of logging to utilize.'),
-                'multiOptions'  => array(
-                    'syslog'    => 'Syslog',
-                    'php'       => $this->translate('Webserver Log', 'app.config.logging.type'),
-                    'file'      => $this->translate('File', 'app.config.logging.type'),
-                    'none'      => $this->translate('None', 'app.config.logging.type')
-                )
+                'multiOptions'  => $types
```
Comment (nilmerg) [inline application/forms/Config/General/LoggingConfigForm.php:57]: Please reduce this complexity monster to this:

```
'value' => Platform::isDocker() ? 'php' : 'syslog'
```

Nothing else than this should be required.

File: application/forms/Config/General/LoggingConfigForm.php [hunk 2]
```diff
@@ -94,7 +105,7 @@ public function createElements(array $formData)
                 )
             );
 
-            if (! isset($formData['logging_log']) || $formData['logging_log'] === 'syslog') {
+            if ((isset($formData['logging_log']) ? $formData['logging_log'] : $defaultType) === 'syslog') {
```
Comment (nilmerg) [inline application/forms/Config/General/LoggingConfigForm.php:108]: Why this change?

File: library/Icinga/Application/Platform.php [hunk 3]
```diff
@@ -59,6 +59,16 @@ public static function isLinux()
...
     }
 
+    /**
+     * Test of Docker
+     *
+     * @return bool
+     */
+    public static function isDocker(): bool
+    {
+        return self::isLinux() && file_exists('/.dockerenv');
```
Comment (nilmerg) [inline library/Icinga/Application/Platform.php:69]: Didn't we talk about env vars? This also returns bool in case of any docker env (including ours) and I don't want these docker related setup changes in my env.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icinga/Application/Platform.php (hunk 3 above)
Comment (Al2Klimov) [reply library/Icinga/Application/Platform.php:69]: Oh. Why not?

File: application/forms/Config/General/LoggingConfigForm.php (hunk 1 above)
Comment (Al2Klimov) [reply application/forms/Config/General/LoggingConfigForm.php]: 😅

File: library/Icinga/Application/Platform.php (hunk 3 above)
Comment (nilmerg) [reply library/Icinga/Application/Platform.php:69]: I'm a developer. I need to test things. If I want to test the setup, I don't want to touch a system file. Additionally, the user and group stuff isn't always the same in every docker setup. **Just in the docker-icingaweb2 image**, that's why.

**Review by nilmerg — APPROVED** [review]
